### PR TITLE
Better Map Activity #365

### PR DESCRIPTION
**Pull request by rohanagarwal94**
Please review my PR and suggest changes if any.
Changes-
1. Better map fragment.
2. Implemented Marker Clustering.
3. Show/Hide markers on various levels in the Indoor Map.
4. AutocompleteTextView for searching places with name.
5. Direction from current location to the marker on click with map toolbar.

Precautions-
1. Please remove my temporary Google Maps API key from Manifest.xml
2. The sample micro locations provided to me by @mananwason is refactored to map.json. I tested the clustering with my custom markers. Try to fill in the LatLong in the JSON file before testing.

### CODE REVIEW

**Comment by the-dagger** [comment]
Hey @rohanagarwal94 could you upload a screenshots depicting the changes? 
Also there are merge conflicts, so could you remove them too? 
Cheers.

File: android/app/build.gradle [hunk 1]
```diff
@@ -85,13 +86,17 @@ dependencies {
     compile 'com.jakewharton.timber:timber:4.1.0'
     compile 'com.jakewharton:butterknife:7.0.1'
 
-    // Googleplay Variant
-    googleplayCompile 'com.google.android.gms:play-services-maps:9.2.0'
+//    // Googleplay Variant
+//    googleplayCompile 'com.google.android.gms:play-services-maps:9.2.0'
```
Comment (mananwason) [inline android/app/build.gradle]: Why remove this?

File: android/app/build.gradle [hunk 2]
```diff
@@ -85,13 +86,17 @@ dependencies {
...
+//    googleplayCompile 'com.google.android.gms:play-services-location:9.2.0'
+
 
     // Fdroid Variant
     fdroidCompile 'org.osmdroid:osmdroid-android:5.1'
     fdroidCompile 'org.apache.directory.studio:org.slf4j.api:1.7.2'
     fdroidCompile 'org.slf4j:slf4j-simple:1.7.14'
+    fdroidCompile 'com.google.android.gms:play-services-maps:9.2.0'
+    fdroidCompile 'com.google.android.gms:play-services-location:9.2.0'
```
Comment (mananwason) [inline android/app/build.gradle]: The Fdroid version is supposed to be without the google maps

File: android/app/src/fdroid/java/org/fossasia/openevent/fragments/MapFragment.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
+package org.fossasia.openevent.fragments;
+
+import android.Manifest;
+import android.content.Context;
```
Comment (mananwason) [inline android/app/src/fdroid/java/org/fossasia/openevent/fragments/MapFragment.java]: This should be done in the Googleplay flavors

File: android/app/src/fdroid/java/org/fossasia/openevent/fragments/OSMapFragment.java [hunk 4]
```diff
@@ -1,153 +1,153 @@
-package org.fossasia.openevent.fragments;
-
-import android.Manifest;
-import android.app.Activity;
```
Comment (mananwason) [inline android/app/src/fdroid/java/org/fossasia/openevent/fragments/OSMapFragment.java]: No need to delete this. This implements the OSM for the fdroid version

**Comment by mananwason** [comment]
@rohanagarwal94 There are a lot of issues in this PR. Please check my comments

**Comment by rohanagarwal94** [comment]
So @mananwason where exactly does the google maps fit in? I thought it was being developed to replace the fdroid map.

**Comment by mananwason** [comment]
@rohanagarwal94 Please read up on gradle flavors. We have two flavors for our project : googlePlay and fdroid. We need better maps for the versions.

**Comment by mananwason** [comment]
I am closing this PR for now.

**Pull request closed without merging**